(maint) Update error maps to be more inline with convention

rlinehan opened the pull request:
Most other projects at PL seem to have standardized on returning a map with
:kind and :msg keys for errors for http responses. This commit brings the
status service in line with that convention.

KevinCorcoran commented:
These changes look fine, as long as it is decided by the Great Powers That Be that this is, in fact, the correct format to use here.

rlinehan commented:
It seems like no one has actually decided whether we should use `type` or `kind`, so I'm going to maybe just change this to remove the nesting and keep the keys as it was before, until we have an actual decision on this.

KevinCorcoran commented:
that works for me.

rlinehan committed:
(maint) Update error maps to be more inline with convention

Our convention in other projects is not to nest the error type/kind and
message underneath an `:error` key. This commit removes that nesting so that
we are more inline with other projects.

rlinehan commented:
updated

KevinCorcoran commented:
:shipit:

MSLilah commented:
:shipit:
